src/transformStateWithClones.js

```diff
@@ -7,7 +7,42 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let prevState = state;
+
+  for (const action of actions) {
+    let newState = transform (prevState, action);
+    result.push(newState);
+    prevState = newState;
+  }
+
+  return result;
+}
+
+function transform (prevState, action) {
+  switch (action.type) {
+    case 'clear':
+      return {};
+
+    case 'addProperties':
+      return {
+        ...prevState,
+        ...action.extraData,
+      };
+
+    case 'removeProperties': {
+      const newState = { ...prevState };
+
+      for (let keys of action.keysToRemove) {
```

Comment:
why let and why plural? you don't change it's value and it's one key
```suggestion
      for (const key of action.keysToRemove) {
```